lib/scoped_search/definition.rb

```diff
@@ -286,7 +288,11 @@ def parse_temporal(value)
       return (eval($1.strip.gsub(/\s+/,'.').downcase)).to_date     if value =~ /\A\s*(\d+\s+\b(?:days?|weeks?|months?|years?)\b\s+\bago)\b\s*\z/i
       return (eval($1.strip.gsub(/from\s+now/i,'from_now').gsub(/\s+/,'.').downcase)).to_datetime if value =~ /\A\s*(\d+\s+\b(?:hours?|minutes?)\b\s+\bfrom\s+now)\b\s*\z/i
       return (eval($1.strip.gsub(/from\s+now/i,'from_now').gsub(/\s+/,'.').downcase)).to_date     if value =~ /\A\s*(\d+\s+\b(?:days?|weeks?|months?|years?)\b\s+\bfrom\s+now)\b\s*\z/i
-      DateTime.parse(value, true) rescue nil
+      if Time.respond_to?(:zone)
+        Time.zone.parse(value)
+      else
+        DateTime.parse(value, true) rescue nil
```

Comment:
We should rescue the `Time.zone` branch too

Reply:
in case of wrong format it does not raise exception but returns nil

Reply:
Not always
```ruby
pry(main)> Time.zone.parse('123:15:00')
ArgumentError: argument out of range
```

Reply:
interesting, updated